app/screens/home/search/results/results.tsx

```diff
@@ -85,13 +143,44 @@ const SearchResults = ({
             );
         }
 
+        if (!fileInfos.length) {
+            return noResults;
+        }
+        const updateFileForGallery = (idx: number, file: FileInfo) => {
+            'worklet';
+            filesForGallery.value[idx] = file;
+        };
+        const container: StyleProp<ViewStyle> = fileInfos.length > 1 ? styles.container : undefined;
+        const isSingleImage = () => (fileInfos.length === 1 && (isImage(fileInfos[0]) || isVideo(fileInfos[0])));
+        const isReplyPost = false;
+
         return (
-            <FileCard
-                fileInfo={item}
+            <View
+                style={container}
                 key={item.id}
-            />
+            >
+                <File
+                    channelName={getChannelName(item.channel_id!)}
+                    galleryIdentifier={galleryIdentifier}
+                    key={item.id}
+                    canDownloadFiles={canDownloadFiles}
+                    file={item}
+                    index={attachmentIndex(item.id!)}
+                    onPress={handlePreviewPress}
+                    onOptionsPress={handleOptionsPress}
+                    theme={theme}
+                    isSingleImage={isSingleImage()}
+                    showDate={true}
+                    publicLinkEnabled={publicLinkEnabled}
+                    updateFileForGallery={updateFileForGallery}
+                    inViewPort={true}
+                    wrapperWidth={(getViewPortWidth(isReplyPost, isTablet) - 6)}
+                    nonVisibleImagesCount={0}
+                    asCard={true}
+                />
+            </View>
         );
-    }, [theme]);
+    }, [theme, fileInfos, fileChannels]);
```

Comment:
You seem to be missing some dependencies, like `publicLinkEnabled`. Some others seem covered, but they are really obscure, like filesForGallery, that I guess it is updated if fileInfos is updated.

Reply:
This information comes directly from an api response to the server. FileChannels and FileInfos will only change when a new search is performed.